libraries/type-length-value/js/src/tlvState.ts

```diff
@@ -0,0 +1,132 @@
+        }
+    }
+
+    private tlvDiscriminatorMatches(type: TlvType, offset: number): boolean {
+        switch (typeof type) {
+            case 'number':
+                return this.readTlvNumberSize(this.typeSize, offset, Number) === type;
+            case 'bigint':
+                return this.readTlvNumberSize(this.typeSize, offset, BigInt) === type;
+            case 'object':
+                if (type instanceof Buffer) {
+                    return this.tlvData.subarray(offset, offset + this.typeSize).equals(type);
+                }
+                if ('bytes' in type) {
+                    return this.tlvData.subarray(offset, offset + this.typeSize).equals(type.bytes);
+                }
+                throw new TlvInvalidDiscriminatorError();
+            default:
+                throw new TlvInvalidDiscriminatorError();
+        }
+    }
+
+    /**
+     * Get a single entry from the tlv data. This function returns the first entry with the given type.
+     *
+     * @param type the type of the entry to get
+     *
+     * @return the entry from the tlv data or null
+     */
+    public firstBytes(type: TlvType): Buffer | null {
+        const entries = this.bytesRepeating(type, 1);
+        return entries.length > 0 ? entries[0] : null;
+    }
+
+    /**
+     * Get a multiple entries from the tlv data. This function returns `count` or less entries with the given type.
+     *
+     * @param type the type of the entry to get
+     * @param count the number of entries to get (0 for all entries)
+     *
+     * @return the entry from the tlv data or null
+     */
+    public bytesRepeating(type: TlvType, count = 0): Buffer[] {
+        const entries: Buffer[] = [];
+        let offset = 0;
+        while (offset < this.tlvData.length) {
+            if (offset + this.typeSize + this.lengthSize > this.tlvData.length) {
+                throw new TlvInvalidAccountSizeError();
+            }
+            const typeMatches = this.tlvDiscriminatorMatches(type, offset);
+            offset += this.typeSize;
+            const entryLength = this.readTlvNumberSize(this.lengthSize, offset, Number);
+            offset += this.lengthSize;
+            if (offset + entryLength > this.tlvData.length) {
+                throw new TlvInvalidAccountSizeError();
+            }
+            if (typeMatches) {
+                entries.push(this.tlvData.subarray(offset, offset + entryLength));
+            }
+            if (count > 0 && entries.length >= count) {
+                break;
+            }
+            offset += entryLength;
+        }
+        return entries;
+    }
+
+    /**
+     * Get all the discriminators from the tlv data. This function will return a type multiple times if it occurs multiple times in the tlv data.
+     *
+     * @return a list of the discriminators.
+     */
+    public discriminators(): Buffer[] {
+        const types: Buffer[] = [];
```

Comment:
Just so it's easier to follow
```suggestion
        const discriminators: Buffer[] = [];
```